### PR TITLE
Proptypes

### PR DESCRIPTION
**Pull request by LayaTaal**
Adds prop type validation to fix lint errors. Apologies for the numerous space/tab fixes included here from the eslint --fix, I hope you are still able to give this a quality review.

I also removed several components that we are no longer using which were throwing prop type errors. I probably should have saved that for another PR, but alas...

Completed #68

### CODE REVIEW

File: components/BlockQuote.js [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
-    </StyledDiv>
+  <StyledDiv>
+    <p>{props.children}</p>
+  </StyledDiv>
 );
 
 export default BlockQuote;
 
+BlockQuote.propTypes = {
+  children: PropTypes.object,
```
Comment (nholden) [inline components/BlockQuote.js]: It looks like `children` is a string in the [one place we use it currently](https://github.com/texas-justice-initiative/website-nextjs/blob/68e583730f93aec6bfc5b9ce293796bcb0a15f3b/pages/about.js#L20-L23). What do you think about making that explicit here?

```suggestion
  children: PropTypes.string.isRequired,
```

File: components/MailChimpForm.js [hunk 2]
```diff
@@ -75,6 +76,12 @@ const CustomForm = ({ status, message, onValidated }) => {
   );
 };
 
+CustomForm.propTypes = {
+  status: PropTypes.string,
+  message: PropTypes.string,
+  onValidated: PropTypes.func,
```
Comment (nholden) [inline components/MailChimpForm.js]: Should we always expect an `onValidated`?

```suggestion
  onValidated: PropTypes.func.isRequired,
```

File: components/Page.js [hunk 3]
```diff
@@ -24,6 +25,10 @@ class Page extends Component {
 
 export default Page;
 
+Page.propTypes = {
+  children: PropTypes.object,
```
Comment (nholden) [inline components/Page.js]: Looks like we're expecting `children` to [be one element right now](https://github.com/texas-justice-initiative/website-nextjs/blob/369fcb1d17f7ce9bbf012ab02b2363af2e104357/pages/_app.js#L22-L26). What do you think about using the doc's suggestion about [requiring a single child](https://reactjs.org/docs/typechecking-with-proptypes.html#requiring-single-child)?

```suggestion
  children: PropTypes.element.isRequired,
```

If we think we'll want to accept multiple children, [StackOverflow](https://stackoverflow.com/questions/42122522/reactjs-what-should-the-proptypes-be-for-this-props-children) seems to prefer `PropTypes.node.isRequired`.

**Comment by LayaTaal** [comment]
Thanks @nholden for the information on element and node proptypes. I'm going to go over this one more time before I finalize this, specifically the `children` prop and try to make it a bit more specific.

**Pull request merged**